Details popup windows

ameerhamzakhalid opened the pull request:
We have implemented the following features:
When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
When the user clicks (or taps) the close (X) button, the popup disappears.
We implement popups for both mobile and desktop screen sizes:

Mobile popup window
Desktop popup window

Lastly, We have fixed Linter errors.

ameerhamzakhalid committed:
Added close button desktop img

I added close button for desktop version

ameerhamzakhalid committed:
Added github white icon

I added white github icon

ameerhamzakhalid committed:
added live icon

I added live icon

ameerhamzakhalid committed:
added popup image

I added popup image for popup window

ameerhamzakhalid committed:
update index.html

We have updated index.html for detail popup windows

ameerhamzakhalid committed:
update index.js

We have updated index.js for detail popup windows

ameerhamzakhalid committed:
update style.css

We have updated style.css for detail popup windows

ameerhamzakhalid committed:
Fixed Linter errors

We have fixed Linter errors

MeqdamAlqudah commented on style.css:
```diff
@@ -297,14 +297,108 @@ button:active {
   justify-content: space-around;
 }
 
-.card-tags > ul > li {
+.card-work > .card-tags > li {
   background: lightgray;
   font-size: 12px;
   padding: 0.5em;
   margin: 0 0.5em;
```
- [ ] Please use the same close button icon as in the Figma design 

| Your design | Figma design |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/74744787/161864381-76fd96b7-0807-4997-969a-c6f08f18938f.png) |  ![image](https://user-images.githubusercontent.com/74744787/161864173-6f3e6641-4a23-46e1-ac56-980a9c48f4a4.png)

MeqdamAlqudah commented on style.css:
```diff
@@ -297,14 +297,108 @@ button:active {
   font-size: 12px;
   padding: 0.5em;
   margin: 0 0.5em;
   border-radius: 2px;
 }
```
- [ ] Please make sure that the popup window background matches the one in the Figma design 

| Your design | Figma design |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/74744787/161864631-6b7429bc-9d8c-4001-abaa-4cb8cc40cadb.png) |  ![image](https://user-images.githubusercontent.com/74744787/161865126-b34ce705-3c85-4963-8efd-6966d25e3db3.png) |

AmaduKamara replied:
- [ ] Kindly work on the popup modal to match the one given in the Figma design just as the previous reviewer requested.

**Hints**
- Work on the width of the modal as it's stretching 100% of the browser window
- Also, please work on the close button. Currently, the close button is not doing what it is supposed to do - close the modal window.
- Kindly make the close button work so when it's clicked, the modal should be closed. 

**This is how your design should look like when the popup shows**

![image](https://user-images.githubusercontent.com/50941074/162048339-c79a0050-6e2b-44ae-898c-ea853be8f897.png)

**Your design**
![image](https://user-images.githubusercontent.com/50941074/162048823-c9e0e4ea-a4df-4fa4-a3ad-a4067175ef25.png)

MeqdamAlqudah commented on index.html:
```diff
@@ -133,7 +133,7 @@ <h2 class="card-title">
+                <button type="button" class="btn-popup">
                     See Project
                 </button>
             </article>
```
- [ ] Please make sure that the popup window is responsive and make sure it looks exactly like the Figma design until 320px you can see from the table below that the background color of the pop window is different also you can see that the margin between the buttons is different than Figma design please fix that

| Your design | Figma design |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/74744787/161865611-6ba876d0-c85a-485f-8f1b-5860afc3c53b.png) | ![image](https://user-images.githubusercontent.com/74744787/161865844-dd319b11-fb6a-496b-8478-ae79651cba32.png) |

MeqdamAlqudah commented on index.js:
```diff
@@ -28,4 +29,159 @@ linkAbout.addEventListener('click', () => {
+for (let i = 0; i < arrayBtnPopup.length; i += 1) {
+  arrayBtnPopup[i].addEventListener('click', () => {
+    document.body.innerHTML = cardWorkDetailsMobile(i);
+    const btnCloseDetailsMobile = document.querySelector('.btn-close-details');
+    btnCloseDetailsMobile.addEventListener('click', async () => {
+      window.location.reload();
+    });
+  });
+}
```
- [ ] Good job Implementing this code to close the popup window but using the async syntax here is useless because the async syntax is made to deal with promises since JavaScript is a single threaded language we need to implement parallel work in the API that why we use aync syntax in some places but here it is useless so I think it would be better to remove it

MeqdamAlqudah reviewed (changes requested):
Hi @jrosario19 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media1.giphy.com/media/neUaH5fz6ZpefE2KBP/200.webp?cid=ecf05e47t9rn9ka6pfrbv06ftvef18dul3p75gwqi4tqlcgq&rid=200.webp&ct=g)

## Highlights

- no linter errors ✅
- readme file ✅
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ameerhamzakhalid committed:
Added close button

I added close button according to figma template as suggested by reviewer

ameerhamzakhalid committed:
Update index.js

I updated close button in index.js as suggested by reviewer.

I removed async syntax in index.js as suggested by reviwer.

ameerhamzakhalid committed:
update style.css

I updated the background color according to figma template suggested by reviewer

AmaduKamara reviewed (changes requested):
# Status: Required Changes ♻️♻️♻️ 

Hi @ameerhamzakhalid, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

```

ameerhamzakhalid committed:
Update CSS

Updated image properties according to suggestions of reviewer

ameerhamzakhalid committed:
Update index.js

I activated close button according to suggestions of reviewer

thecodechaser commented on index.html:
```diff
@@ -116,7 +116,7 @@ <h2 class="card-title">
                     <li>JavaScript</li>
                     <li>html</li>
                 </ul>
```
- [ ] [ optional ] You need to update the main page so that the projects section is created dynamically using the information stored in that JavaScript object. Remember, all of the HTML in that section is created when the page loads. Please update the projects section from the `index.html` file and make sure that section is created using javascript dynamically.

### See the image of missing requirements:

![image](https://user-images.githubusercontent.com/83803414/161791278-f45b9334-bf67-4132-869a-74bebdbae133.png)

ameerhamzakhalid committed:
updated index.js

ameerhamzakhalid committed:
update style.css

thecodechaser reviewed (approved):
## STATUS: APPROVED 🟢

## Hi @ameerhamzakhalid👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media3.giphy.com/media/g9582DNuQppxC/giphy-downsized-medium.gif"/>

## Optional suggestions 
#### I am marking these changes as optional because they were missed by the previous reviewers.

- [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
